src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/config/WebClientConfiguration.kt

```diff
@@ -1,71 +1,71 @@
 
   @Bean
-  fun mappingApiHealthWebClient(): WebClient = WebClient.builder().baseUrl(mappingApiBaseUri).build()
+  fun mappingApiHealthWebClient(builder: WebClient.Builder): WebClient = builder.healthWebClient(mappingApiBaseUri, healthTimeout)
 
   @Bean
-  fun oauthApiHealthWebClient(): WebClient = WebClient.builder().baseUrl(oauthApiBaseUri).build()
+  fun oauthApiHealthWebClient(builder: WebClient.Builder): WebClient = builder.healthWebClient(oauthApiBaseUri, healthTimeout)
 
   @Bean
-  fun nomisApiWebClient(authorizedClientManager: ReactiveOAuth2AuthorizedClientManager, webClientBuilder: WebClient.Builder): WebClient {
-    val oauth2Client = ServerOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
-
-    oauth2Client.setDefaultClientRegistrationId("nomis-api")
-
-    return webClientBuilder
-      .baseUrl(nomisApiBaseUri)
-      .filter(oauth2Client)
-      .build()
-  }
+  fun nomisApiWebClient(authorizedClientManager: ReactiveOAuth2AuthorizedClientManager, builder: WebClient.Builder): WebClient =
+    builder.authorisedWebClient(authorizedClientManager, registrationId = "nomis-api", url = nomisApiBaseUri, timeout)
 
   @Bean
-  fun mappingApiWebClient(authorizedClientManager: ReactiveOAuth2AuthorizedClientManager): WebClient {
-    val oauth2Client = ServerOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
-    oauth2Client.setDefaultClientRegistrationId("visit-mapping-api")
-
-    return WebClient.builder()
-      .baseUrl(mappingApiBaseUri)
-      .filter(oauth2Client)
-      .build()
-  }
+  fun mappingApiWebClient(authorizedClientManager: ReactiveOAuth2AuthorizedClientManager, builder: WebClient.Builder): WebClient =
+    builder.authorisedWebClient(authorizedClientManager, registrationId = "nomis-mapping-api", url = mappingApiBaseUri, timeout)
 
   @Bean
   fun authorizedClientManager(
     clientRegistrationRepository: ReactiveClientRegistrationRepository,
     oAuth2AuthorizedClientService: ReactiveOAuth2AuthorizedClientService,
   ): ReactiveOAuth2AuthorizedClientManager {
     val authorizedClientProvider = ReactiveOAuth2AuthorizedClientProviderBuilder.builder().clientCredentials().build()
-    val authorizedClientManager =
-      AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager(clientRegistrationRepository, oAuth2AuthorizedClientService)
-    authorizedClientManager.setAuthorizedClientProvider(authorizedClientProvider)
-    return authorizedClientManager
+    return AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager(
+      clientRegistrationRepository,
+      oAuth2AuthorizedClientService,
+    ).kotlinApply { setAuthorizedClientProvider(authorizedClientProvider) }
+  }
+}
+
+fun WebClient.Builder.authorisedWebClient(authorizedClientManager: ReactiveOAuth2AuthorizedClientManager, registrationId: String, url: String, timeout: Duration): WebClient {
```

Comment:
switched to extension functions so don't need inheritance for all the client configurations